src/main/java/com/netflix/conductor/kafka/eventqueue/KafkaObservableQueue.java

```diff
@@ -0,0 +1,408 @@
+            if (queueNameParts.length > 1) {
+                this.groupId = queueNameParts[0];
+                this.queueName = queueNameParts[1];
+            } else {
+                this.queueName = queueNameParts[0];
+            }
+        } else {
+            this.queueName = queueParams.get(QUEUE_PARAM_TOPICS);
+            if (queueParams.containsKey(QUEUE_PARAM_GROUP)) {
+                this.groupId = queueParams.get(QUEUE_PARAM_GROUP);
+            }
+            if (queueParams.containsKey(QUEUE_PARAM_FILTER_NAME) && queueParams.containsKey(QUEUE_PARAM_FILTER_VALUE)) {
+                String filterName = queueParams.get(QUEUE_PARAM_FILTER_NAME);
+                if (QUEUE_FILTER_BY_PARTITION_KEY.equals(filterName)) {
+                    this.filterByPartitionKey = true;
+                } else {
+                    this.filteringHeader = filterName;
+                }
+                this.filteringValue = queueParams.get(QUEUE_PARAM_FILTER_VALUE);
+            }
+        }
+        this.topics = Arrays.asList(this.queueName.split(","));
+
+        if (properties.getPollIntervalMs() != null) {
+            this.pollIntervalInMS = properties.getPollIntervalMs();
+        }
+        if (properties.getPollTimeoutMs() != null) {
+            this.pollTimeoutInMs = properties.getPollTimeoutMs();
+        }
+        this.producer = producer;
+        this.objectMapper = new ObjectMapper();
+
+        init(properties);
+    }
+
+    private Map<String, String> parseParams() {
+        Matcher matcher = QUEUE_PARAMS_PATTERN.matcher(this.queueURI);
+        Map<String, String> params = new HashMap<>();
+        while (matcher.find()) {
+            params.put(matcher.group(1), matcher.group(2));
+        }
+        if (!params.containsKey(QUEUE_PARAM_TOPICS)) {
+            return new HashMap<>();
+        }
+        return params;
+    }
+
+    /**
+     * Initializes the kafka producer with the defaults. Fails in case of any
+     * mandatory configs are missing.
+     *
+     * @param properties
+     */
+    private void init(KafkaEventQueueProperties properties) {
+        try {
+            Properties consumerProperties = new Properties();
+            String prop = properties.getBootstrapServers();
+            if (prop != null) {
+                consumerProperties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, prop);
+            }
+            prop = this.groupId;
+            if (prop != null) {
+                consumerProperties.put(ConsumerConfig.GROUP_ID_CONFIG, prop);
+            }
+            prop = properties.getOffset();
+            if (prop != null) {
+                consumerProperties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, prop);
+            }
+            //consumerProperties.put("security.protocol", SecurityProtocol.SASL_PLAINTEXT.name);
+            consumerProperties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
```

Comment:
I guess we should use ByteArrayDeserializer here as messages are going to be filter out based of headers info and in this case we don't need to serialize data at all.

Reply:
We need to deserialize key for filtering according to jira issue

Reply:
filtering is done based on headers value. We don't use "data" for filtering

Reply:
I agree, but you selected the KEY deserializer code.